### PR TITLE
Add favourites screen and contexts

### PR DESCRIPTION
**Pull request by UtsavK55**
This PR consists of favourites screen, user and date contexts and changes related to it

### CODE REVIEW

File: src/contexts/DateProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import {createContext, ReactNode, useContext, useState} from 'react';
+
+export const DateContext = createContext<DateContext | undefined>(undefined);
+
+export const useDateContext = () => {
+  const context = useContext(DateContext);
+  if (!context) {
+    throw new Error('userContext must be used within a DateProvider');
```
Comment (02priyanshi) [inline src/contexts/DateProvider.tsx]: why are you throwing error of userContext in date context

File: src/screens/favourites/Favourites.tsx [hunk 2]
```diff
@@ -1,11 +1,79 @@
...
+import RecipeCard from '@components/recipeCard';
+import NoDataFound from '@components/noDataFound';
+import {ROUTES, STORAGE_KEYS} from '@constants';
+import {getRecipeByIdUrl} from '@network/apiUrl';
+import {fetchData} from '@network/apiMethods';
+import {getData} from '@storage';
+import {useThemeColors} from '@theme';
+
+import {favoritesScreenStyles} from './styles';
 import BaseContainer from '@components/baseContainer';
```
Comment (02priyanshi) [inline src/screens/favourites/Favourites.tsx]: import order

File: src/screens/favourites/Favourites.tsx [hunk 3]
```diff
@@ -1,11 +1,79 @@
...
+    );
+    setFavorites(favoriteRecipes);
+  }, [isFocused]);
+
+  useEffect(() => {
+    getFavorites();
+  }, [getFavorites]);
+
+  const handleUnfavorite = useCallback(() => {
+    getFavorites();
```
Comment (02priyanshi) [inline src/screens/favourites/Favourites.tsx]: why are you calling same function again

**Review by 02priyanshi — CHANGES_REQUESTED** [review]

File: src/screens/favourites/Favourites.tsx (hunk 3 above)
Comment (UtsavK55) [reply src/screens/favourites/Favourites.tsx]: To get the latest changes on marking it unfavourite

Comment (02priyanshi) [reply src/screens/favourites/Favourites.tsx]: not a correct way change approach

**Review by 02priyanshi — APPROVED** [review]

**Pull request merged**